extension/observer/ecsobserver/exporter.go

```diff
@@ -14,10 +14,116 @@
 
 package ecsobserver
 
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultMetricsPath = "/metrics"
+)
+
 // CommonExporterConfig should be embedded into filter config.
 // They set labels like job, metrics_path etc. that can override prometheus default.
 type CommonExporterConfig struct {
 	JobName      string `mapstructure:"job_name" yaml:"job_name"`
 	MetricsPath  string `mapstructure:"metrics_path" yaml:"metrics_path"`
 	MetricsPorts []int  `mapstructure:"metrics_ports" yaml:"metrics_ports"`
 }
+
+// taskExporter converts annotated Task into PrometheusECSTarget.
+type taskExporter struct {
+	logger  *zap.Logger
+	cluster string
+}
+
+func newTaskExporter(logger *zap.Logger, cluster string) *taskExporter {
+	return &taskExporter{
+		logger:  logger,
+		cluster: cluster,
+	}
+}
+
+// ExportTasks loops a list of tasks and export prometheus scrape targets.
+// It keeps track of error but does NOT stop when error occurs.
+// The returned targets are all valid and the error(s) are mainly for generating metrics.
+func (e *taskExporter) ExportTasks(tasks []*Task) ([]PrometheusECSTarget, error) {
```

Comment:
It might not matter as much but for clarity let's make methods private too for private types

Reply:
done